src/consensus.rs

```diff
@@ -399,8 +400,14 @@ impl Consensus {
 
     /// Listens for the next proposal and sends it to the Raft node.
     fn propose_updates(&mut self, timeout: Duration) -> anyhow::Result<()> {
-        match self.receiver.recv_timeout(timeout) {
-            Ok(Message::FromPeer(message)) => {
+        // Poll the async. channel on the consensus runtime.
+        // https://docs.rs/tokio/1.22.0/tokio/sync/mpsc/index.html#communicating-between-sync-and-async-code
+        let received = self.runtime.block_on(async {
+            // Wait for the next proposal during `timeout`.
+            tokio::time::timeout(timeout, self.receiver.recv()).await
+        });
```

Comment:
```suggestion
        let received = self.runtime.block_on(tokio::time::timeout(
            // Wait for the next proposal during `timeout`.
            timeout,
            self.receiver.recv(),
        ));
```
It actually works without async block as well. Checked with clippy

Reply:
Oh, that's a runtime error. Then you are right, sorry

Reply:
No worries, I found out about this runtime issue via our unit tests :)